distribution/lib/Standard/Base/0.0.0-dev/src/Internal/Time/Format/Parser.enso

```diff
@@ -0,0 +1,270 @@
+                        Panic.throw (Date_Time_Format_Parse_Error.Error "The curly braces setting the maximum year for `yy` must be an integer, but got: {"+parameter_text+"}.")
+            Two_Digit_Year_Representation.Value max_year=max_year
+        _ ->
+            Numeric_Representation.Value count
+
+    ## PRIVATE
+    parse_simple_date_pattern self character count =
+        lowercase = character.to_case Case.Lower
+        case lowercase of
+            "y" -> Standard_Date_Patterns.Year (self.resolve_year_representation count)
+            "q" -> Standard_Date_Patterns.Quarter (Numeric_Representation.Value count)
+            "m" -> case character of
+                "M" ->
+                    representation = case count of
+                        1 -> Numeric_Representation.Value 1
+                        2 -> Numeric_Representation.Value 2
+                        3 -> Text_Representation.Short_Form
+                        _ -> Text_Representation.Long_Form
+                    Standard_Date_Patterns.Month representation
+                # Lowercase form is reserved for minutes - handled elsewhere.
+                "m" -> Nothing
+            "d" -> case count of
+                1 -> Standard_Date_Patterns.Day_Of_Month (Numeric_Representation.Value 1)
+                2 -> Standard_Date_Patterns.Day_Of_Month (Numeric_Representation.Value 2)
+                3 -> Standard_Date_Patterns.Day_Of_Week (Text_Representation.Short_Form)
+                _ -> Standard_Date_Patterns.Day_Of_Week (Text_Representation.Long_Form)
+            "e" -> self.fail_invalid_pattern character count " If you want to represent the day of week as text use `ddd` for the short form and `dddd` for the long form, or use `from_java` for the Java syntax."
+            "l" -> self.fail_invalid_pattern character count " If you want to represent the month as text use `MMM` for the short form and `MMMM` for the long form, or use `from_java` for the Java syntax."
+            _ -> Nothing
+
+    ## PRIVATE
+    parse_iso_week_year_pattern self character count =
+        lowercase = character.to_case Case.Lower
+        case lowercase of
+            "y" -> ISO_Week_Year_Patterns.Week_Based_Year (self.resolve_year_representation count)
+            "w" -> ISO_Week_Year_Patterns.Week_Of_Year (Numeric_Representation.Value count)
+            "d" -> case count of
+                1 -> ISO_Week_Year_Patterns.Day_Of_Week (Numeric_Representation.Value 1)
+                2 -> ISO_Week_Year_Patterns.Day_Of_Week (Numeric_Representation.Value 2)
+                3 -> ISO_Week_Year_Patterns.Day_Of_Week (Text_Representation.Short_Form)
+                _ -> ISO_Week_Year_Patterns.Day_Of_Week (Text_Representation.Long_Form)
+            "e" -> self.fail_invalid_pattern character count " If you want to represent the day of week as text use `ddd` for the short form and `dddd` for the long form, or use `from_java` for the Java syntax."
```

Comment:
slightly tempted to allow `e` to be `ddd` or `ee+` to `dddd` in this specialised mode but not a big thing.

Reply:
I think that is a good idea indeed, for both modes even.